DEV_GUIDE.md

```diff
@@ -0,0 +1,214 @@
+# Development guide
+
+This guide explains how to modify or contribute to the Forjj  project.
+Forjj is written in GO language, that needs to be compiled. This is the main difficulty, however some scripts have been written to really simplify this process.
+
+
+## Repositories
+
+To properly build Forjj you need to get the sources from github.
+
+Forjj is composed of various repositories that are available from the project main page https://github.com/forj-oss .
+- forjj --> main program, cli
+- forjj-contribs --> plugins (github, jenkins)
+- forjj-flows --> ?
+- forjj-modules --> Modules used by Forjj cli
+- forjj-repotemplates --> Templates for created project by Forjj
+- goforjj --> GO templates in order to build plugins
```

Comment:
It defines/centralizes also the REST API interface between forjj and plugins written in GO.